admin/controllers/school.js

```diff
@@ -439,62 +400,39 @@ const savePupilNotTakingCheck = async (req, res, next) => {
     return res.redirect('/school/pupils-not-taking-check/select-pupils')
   }
 
-  const todayDate = moment(moment.now()).format()
-  const postedPupils = req.body.pupil
-  const pupilsData = await pupilService.fetchMultiplePupils(Object.values(postedPupils))
-
-  let pupilsList
-  let attendanceCodes
-  let pupils
-
-  for (let index = 0; index < pupilsData.length; index++) {
-    let pupil = pupilsData[index]
-    pupil.attendanceCode = {
-      _id: req.body.attendanceCode,
-      dateRecorded: new Date(todayDate),
-      byUserName: req.user.UserName,
-      byUserEmail: req.user.EmailAddress
+  // The req.body.pupil data is posted in 3 forms:
+  // 1: string: 'abc-def' (single selection)
+  // 2: array of strings: ['abc-def', 'foo-bar'] (multiple selection)
+  // 3: object with properties/values: { 0: 'abc-def, 1: 'foo-bar' } (using checkbox "Select all")
+  let postedPupilSlugs
+  if (typeof req.body.pupil === 'object') {
```

Comment:
how come there's an inconsistency on the data type of pupil coming from the frontend>

Reply:
String versus array depends on whether you pick one or more pupils.  I don’t know why it sometimes gets turned into an object.

Reply:
We'll definitely need to investigate this, it's very obnoxious.